src/MoBi.Presentation/MenusAndBars/ContextMenus/ContextMenuForModule.cs

```diff
@@ -43,10 +43,20 @@ public IContextMenu InitializeWith(ObjectBaseDTO dto, IPresenter presenter)
          _allMenuItems.Add(createSaveItemFor(module));
          _allMenuItems.Add(createRemoveItemFor(module));
          _allMenuItems.Add(createCloneMenuItem(module));
+
+         if(module.HasSnapshot)
+            _allMenuItems.Add(createRecreateMenuItem(module));
          
          return this;
       }
 
+      private IMenuBarItem createRecreateMenuItem(Module module)
+      {
+         return CreateMenuButton.WithCaption(AppConstants.MenuNames.RecreateFromSnapshot)
+            .WithCommandFor<RecreateModuleUICommand, Module>(module, _container)
```

Comment:
what if we keep with the nomenclature LoadFromSnapshot? what's with the "Recreate" lol